### ISSUE TITLE
`jnp.bfloat16`'s `smallest_subnormal` truncated to zero when converted to `jnp.float64` and then back to `jnp.bfloat16`.

### ISSUE THREAD

**Issue by gonnet** [jax-ml/jax#38995]
### Description

I have the following function:
```py
  jax.config.update('jax_enable_x64', True)  # For `jnp.float64`.

  def trunc_to_bfloat16(values: jax.Array) -> jax.Array:
    return values.astype(jnp.bfloat16).astype(values.dtype)
```
which takes a `jax.Array`, casts it to `jnp.bfloat16`, and then back to its original `dtype`.

The following works as expected:
```py
  ssn = jnp.finfo(jnp.bfloat16).smallest_subnormal
  assert trunc_to_bfloat16(jnp.asarray(ssn)).item() == ssn
  assert trunc_to_bfloat16(jnp.asarray(ssn, jnp.bfloat16)).item() == ssn
  assert trunc_to_bfloat16(jnp.asarray(ssn, jnp.float32)).item() == ssn
```
The following, however, does not:
```py
  assert trunc_to_bfloat16(jnp.asarray(ssn, jnp.float64)).item() == ssn
```
i.e. `trunc_to_bfloat16(jnp.asarray(ssn, jnp.float64)).item()` returns `0.0`. Note that it's not the conversion `jnp.asarray(ssn, jnp.float64)` as the following works:
```py
  assert jnp.asarray(ssn, jnp.float64).item() == ssn
```

### System info (python version, jaxlib version, accelerator, etc.)

```
jax:    0.11.0
jaxlib: 0.11.0
numpy:  2.4.6
python: 3.13.11 (stable, redacted, redacted)
device info: cpu-1, 1 local devices"
process_count: 1
platform: uname_result(system='Linux', node='**redacted**', release='6.18.14-1rodete3-amd64', version='#1 SMP PREEMPT_DYNAMIC Debian 6.18.14-1rodete3 (2026-03-30)', machine='x86_64')
```

**Comment by jakevdp** [comment]
In general, XLA uses [flush to zero semantics](https://docs.jax.dev/en/latest/notebooks/Common_Gotchas_in_JAX.html#miscellaneous-divergences-from-numpy) for subnormals in most operations, so this behavior is expected.

**Comment by gonnet** [comment]
> In general, XLA uses flush to zero semantics for subnormals in most operations, so this behavior is expected.

Granted, but then why does this work for `jnp.float32` and `jnp.bfloat16`, but not for `jnp.float64`?

It's particularly weird since `ssn` is perfectly representable in `jnp.float64`, but is a subnormal in both `jnp.float32` and `jnp.bfloat16`, so I'd argue it's not really expected given that behaviour...

**Comment by jakevdp** [comment]
> Granted, but then why does this work for `jnp.float32` and `jnp.bfloat16`, but not for `jnp.float64`?

converting between `float32` and `bfloat16` is simply a matter of truncating or padding mantissa bits in the float representation, so for efficiency it does not involve any value-dependent logic. Converting `bfloat16` to `float64` and back is more involved (since `float64` has more bits allocated to the exponent of the floating point representation), and so flush-to-zero semantics for subnormals come into play.

**Comment by CodersAcademy006** [comment]
Reproduced on CPU (jax 0.10.2, x64 enabled). The truncation is specifically in the `float64 -> bfloat16` conversion, not the round trip in general:

```python
ssn = float(jnp.finfo(jnp.bfloat16).smallest_subnormal)  # 9.18e-41
jax.lax.convert_element_type(jnp.asarray(ssn, jnp.float64), jnp.bfloat16)  # -> 0.0
jax.lax.convert_element_type(jnp.asarray(ssn, jnp.float32), jnp.bfloat16)  # -> ssn (correct)
```

`float32 -> bfloat16` keeps the subnormal, and `numpy`/`ml_dtypes` do the `float64 -> bfloat16` cast correctly (`float(ml_dtypes.bfloat16(np.float64(ssn))) == ssn`), so the issue is isolated to XLA's `ConvertElementType` lowering for `f64 -> bf16`, which appears to round toward zero instead of round-to-nearest for a value that is exactly the smallest bf16 subnormal. That also explains why every other source dtype in the report works: they either go through the correct `f32 -> bf16` path or don't involve the `f64 -> bf16` convert. This looks like a jaxlib/XLA fix rather than something in the Python layer.

**Comment by jakevdp** [comment]
Closing: this is working as intended.
